### ISSUE TITLE
[Wiki] takes more than 4 seconds to complete loading

### ISSUE THREAD

**Issue by rassi0429** [Yellow-Dog-Man/Resonite-Issues#1130]
### Is your feature request related to a problem? Please describe.

Thank you for hosting the Wiki!
Currently, when accessing from Japan, it takes more than 4 seconds to complete loading. 
The Wiki appears to be hosted in Finland, which is physically far from Japan. there are times when it can't be helped.

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/91118218/63e9180d-ede0-4f2e-91b0-d0a5f38ad5c1)

### Describe the solution you'd like

Proxy Wiki with a CDN such as Cloudflare. Alternatively,they can proxy only static assets.
This improves load times by generating cache in CloudFlare data centers around the world.

### Describe alternatives you've considered

Basically CloudFlare should be fine to proxy all requests. It also offers SSL and DDoS protection, which should be a big advantage.

**Comment by ProbablePrime** [comment]
I did some research on this today. It turns out this is totally possible but complicated.

I'll drop my notes below:
- [ ] Set: https://www.mediawiki.org/wiki/Manual:$wgCdnServers to have all Cloudflare IPs
- [ ] Set: https://www.mediawiki.org/wiki/Manual:$wgUseCdn to true
- [ ] Enable: https://developers.cloudflare.com/support/troubleshooting/restoring-visitor-ips/restoring-original-visitor-ips on wiki webserver
- [ ] Install: https://www.mediawiki.org/wiki/Extension:Cloudflare and set to just file mode for now
- [ ] install: https://www.mediawiki.org/wiki/Extension:CloudflarePurge and configure as normal
	- Why both? - BECAUSE OSS IS FUN
- [ ] consult: https://www.mediawiki.org/wiki/Manual:Cloudflare for additional stuff
- [ ] Only when all of the above is done, proxy cloudflare on cloudflare's dashboard.

**Comment by ProbablePrime** [comment]
This may be partially better now, Images/Other files now go over CF CDN, So give it a go. I'll still be working on my list above later.
